README.md

```diff
@@ -60,17 +61,96 @@ project(':react-native-uport-signer').projectDir = new File(rootProject.projectD
 4. Open up `android/app/src/main/java/[...]/MainActivity.java`
   - Add `import com.reactlibrary.RNUportSignerPackage;` to the imports at the top of the file
   - Add `new RNUportSignerPackage()` to the list returned by the `getPackages()` method
-  
+
+
 5. [optional] If not already done, update the `minSdkVersion` of your `app` to 21
   This is usually in `android/app/build.gradle` but can also be defined in `android/build.gradle`
 depending on when your project was created
 
 
-## Usage
+## HD Signer Usage
 ```javascript
-import RNUportSigner from 'react-native-uport-signer';
+import { RNUportHDSigner } from 'react-native-uport-signer';
+
+// Create Seed
+RNUportHDSigner.createSeed('simple').then( seed => {
+	console.log(seed.address)
+  console.log(seed.pubKey)
+})
+
+// Get the seed phrase
+RNUportHDSigner.showSeed(this.state.address, 'simple').then (seed => {
+	console.log(seed)
+})
+
+// Delete seed
+RNUportHDSigner.deleteSeed(this.props.address)
+
+// Import Seed
+RNUportHDSigner.importSeed(seedPhrase, 'simple').then(addressObj => {
+	console.log(addressObj.address)
+	console.log(addressObj.pubKey)
+})
+
+//Derive another Address
+RNUportHDSigner.addressForPath(this.state.address, `m/7696500'/0'/1'/0'`,'prompt').then (seed => {
+	console.log(seed)
+})
+
+// Signing a JWT 
+RNUportHDSigner.signJwt(address,
+        RNUportHDSigner.UPORT_ROOT_DERIVATION_PATH,
+        base64EncodedJwt,
+        'simple'
+    ).then( jwtSig => {
+			console.log(jwtSig.r)
+			console.log(jwtSig.s)
+			console.log(jwtSig.v)
+		})
+		
+// Signing an Eth Tx
+RNUportHDSigner.signTx(this.props.address,
+        RNUportHDSigner.UPORT_ROOT_DERIVATION_PATH,
+        base64Tx,
+        'simple'
+    ).then( sig => {
+			console.log(sig.r)
+			console.log(sig.s)
+			console.log(sig.v)
+    })
+```
 
-// TODO: What to do with the module?
-RNUportSigner;
+## Basic Signer Usage
+```javascript
+import { RNUportSigner } from 'react-native-uport-signer';
+
+// Create a keypair
+RNUportSigner.createKeyPair('simple').then(keypair => {
+			console.log(keypair.address)
+			console.log(keypar.pubKey)
+})
+
+//Sign a JWT
+const exampleJwtPayload = { iss: this.props.address, aud: this.props.address, name: 'test'}
+
+RNUportSigner.signJwt(this.props.address,
```

Comment:
`this.props.address` is highly specific and can induce confusion.
I think `address` is enough